tools/pylib/boututils/check_scaling.py

```diff
@@ -0,0 +1,70 @@
+"""Functions for checking the error scaling of MMS or MES results
+
+"""
+
+from numpy import array, isclose, log, polyfit
+
+
+def get_order(grid_spacing, errors):
+    """Get the convergence order of errors over the full range of
+    grid_spacing, and at small spacings
+
+    Parameters
+    ----------
+    grid_spacing : list of float
+        The grid spacing or inverse of number of grid points
+    errors : list of float
+        The error at each grid spacing
+
+    Returns
+    -------
+    tuple of float
+        The first value is the error scaling over the full range of
+        grid spacings; the second value is the scaling over the last
+        two points
+
+    """
+    full_range = polyfit(log(grid_spacing), log(errors), 1)
```

Comment:
Wouldn't a point-by-point comparison be the strictest?
I would prefer if that would be the default way to check the error.

Reply:
That might be strictest way, but that doesn't give you a single value for the order. We could take the average, but that would be prone to the same errors as fitting.